File: Modules/Data/Data.Shared/Events.cs
using System;
using Geo.Geometries;
using Trackwane.Framework.Common;

// ReSharper disable CheckNamespace
namespace Trackwane.Data.Shared.Events
// ReSharper restore CheckNamespace
{
    public class SensorReadingSaved : DomainEvent
    {
        public string ReadingKey { get; set; }
        public string TrackerKey { get; set; }
        public string OrganizationKey { get; set; }
        public string HardwareId { get; set; }
        public Point Coordinates { get; set; }
        public DateTime? Timestamp { get; set; }
        public double? Speed { get; set; }
        public double? Orientation { get; set; }
        public double? Petrol { get; set; }
        public double? BatteryLevel { get; set; }
        public double? Heading { get; set; }
        public decimal? Distance { get; set; }
    }
}

File: Modules/Data/Data.Contracts/Events.fs
namespace Trackwane.Data.Events

open System
open Trackwane.Framework.Common
open Geo.Geometries

type SensorReadingSaved() = 
    inherit DomainEvent()
    
    member val ReadingKey : string = null with get, set
    
    member val TrackerKey : string = null with get, set
    
    member val OrganizationKey : string = null with get, set
    
    member val HardwareId : string = null with get, set
    
    member val Coordinates : Point = null with get, set
    
    member val Timestamp : Nullable<DateTime> = Nullable() with get, set
    
    member val Speed : Nullable<double> = Nullable() with get, set
    
    member val Orientation : Nullable<double> = Nullable() with get, set
    
    member val Petrol : Nullable<double> = Nullable() with get, set
    
    member val BatteryLevel : Nullable<double> = Nullable() with get, set
    
    member val Heading : Nullable<double> = Nullable() with get, set

    member val Distance : Nullable<decimal> = Nullable() with get, set

   

       

        